icu4c/source/common/loclikelysubtags.cpp

```diff
@@ -565,24 +565,16 @@ LSR LikelySubtags::makeMaximizedLsr(const char *language, const char *script, co
     // They should match only themselves,
     // not other locales with what looks like the same language and script subtags.
     char c1;
-    if (region[0] == 'X' && (c1 = region[1]) != 0 && region[2] == 0) {
+    if (region[0] == 'X' && (c1 = region[1]) != 0 && region[2] == 0 &&
```

Comment:
This does not seem to fix the `PS*` variants.

Just wrapping both `if`s seems to do the job:
```java
if (!returnInputIfUnmatch) {
    if (region[0] == 'X' && (c1 = region[1]) != 0 && region[2] == 0) {
        ...
    }
    if (variant[0] == 'P' && variant[1] == 'S') {
        ...
    }
}
```

Reply:
Same applies to icu4c, of course